x-pack/plugins/security/public/plugin.ts

```diff
@@ -19,6 +19,7 @@ export class SecurityPlugin implements Plugin<SecurityPluginSetup, SecurityPlugi
     anonymousPaths.register('/login');
     anonymousPaths.register('/logout');
     anonymousPaths.register('/logged_out');
+    anonymousPaths.register('/status');
```

Comment:
I had to add the status page in the list of anonymous paths, else we were using the authenticated url for capabilities call, therefor returning a 401 redirecting to the login page. Ideally this will move to the actual status_page plugin once migrated, but for now, as legacy plugins are invoked after core's `start` phase (therefor too late for the capabilities call), I had to put it in a NP plugin, and there looked like the best place.

Reply:
All pages in this list are `hidden/chromeless` apps. Do they utilize capabilities for real? Probably we should fetch capabilities only when we have real consumers/subscribers?

Reply:
I wanted to do that. However, we make no distinction between apps, being `hidden`, `anonymous` or `classic`. That means we start every enabled plugins on any page. And plugin currently makes the assumption that capabilities are loaded.

Not fetching proper model result in runtime errors, as plugins assume they are fetched and makes no null check (i.e `newPlatform.core.applications.capabilities.mycustompluginsection.myproperty` which will error on `mysection is undefined`)

To be honest there is a lot we could (and that I want to) cleanup regarding capabilities on client side:
- rename this `navLink` to `application` (navLink is a nonsense in NP)
- restrict access to plugin to their specific capabilities
- find a way to enforce capabilities types (its currently a Record of Record...)

 but I see this more post 8.0 when every plugin is in TS and that we can rely on the compiler for deep refactors and type usages). Main goal of this PR was to allow NP plugin to register capabilities, as it was currently only possible from legacy.

Reply:
could you create an issue for the improvements?

Reply:
https://github.com/elastic/kibana/issues/51956 created.